File: wordcount/wordcount.py
#!/usr/bin/python2
# vim: set fileencoding=utf8 :

import sys
from pyspark import SparkContext

sc = SparkContext()

if len(sys.argv) < 3:
    print("no input file specified")
    sys.exit(1)

inputFile = sys.argv[1]
outFile = sys.argv[2]

print("input %s" % inputFile)

file = sc.textFile(inputFile)
counts = file.flatMap(lambda line: line.split(" ")) \
             .map(lambda word: (word, 1)) \
             .reduceByKey(lambda a, b: a + b)
counts.saveAsTextFile(outFile)

File: wordcount/scala/wordcount.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object WordCount {
  def main(args: Array[String]) {

    val sc = new SparkContext(new SparkConf().setAppName("Spark Count"))

    if(args.length < 2){
      println("No input file provided")
      System.exit(1)
    }
    val file = sc.textFile(args(0))
    val counts = file.flatMap(line => line.split(" "))
                     .map(word => (word, 1))
                     .reduceByKey(_ + _)
    counts.saveAsTextFile(args(1))

    }

  }
